signal.go

```diff
@@ -14,6 +14,12 @@ type waiter struct {
 	minState bool
 }
 
+// StateReader is a read only abstraction of a Signal to expose the current state.
+type StateReader interface {
+	State() State
+	WaitForState(state State) <-chan struct{}
```

Comment:
What about the other functions? I know they're mostly duplication, but it's convenient :). And if we have them anyway, why not expose them in the reader interface?
I mean `IsState` and `WaitForStateMin` and `ObserveStateChange`

Reply:
Good idea + maybe we can change the places where we read/wait/watch state (if any) to use the reader and not access the state directly (?)